components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/oauth2/OAuthConsumerAppToApiModel.java

```diff
@@ -146,4 +156,43 @@ private List<String> getCallbackUrls(OAuthConsumerAppDTO oauthApp) {
         }
         return callbackUris;
     }
+
+    private ClientAuthenticationConfiguration buildClientAuthenticationConfiguration(OAuthConsumerAppDTO appDTO) {
+
+        return new ClientAuthenticationConfiguration()
+                .tokenEndpointAuthMethod(appDTO.getTokenEndpointAuthMethod())
+                .tokenEndpointAuthSigningAlg(appDTO.getTokenEndpointAuthSignatureAlgorithm())
+                .tlsClientAuthSubjectDn(appDTO.getTlsClientAuthSubjectDN());
+    }
+
+    private RequestObjectConfiguration buildRequestObjectConfiguration(OAuthConsumerAppDTO oAuthConsumerAppDTO) {
+
+        return new RequestObjectConfiguration()
+                .requestObjectSigningAlg(oAuthConsumerAppDTO.getRequestObjectSignatureAlgorithm())
+                .encryption(buildRequestObjectEncryptionConfiguration(oAuthConsumerAppDTO));
+    }
+
+    private RequestObjectEncryptionConfiguration buildRequestObjectEncryptionConfiguration(OAuthConsumerAppDTO appDTO) {
+
+        String algorithm = appDTO.getRequestObjectEncryptionAlgorithm();
+        String method = appDTO.getRequestObjectEncryptionMethod();
+        return new RequestObjectEncryptionConfiguration()
+                .algorithm(StringUtils.equals(algorithm, "null") ||
+                        StringUtils.isBlank(algorithm) ? "" : appDTO.getRequestObjectEncryptionAlgorithm())
+                .method(StringUtils.equals(method, "null") ||
+                        StringUtils.isBlank(method) ? "" : appDTO.getRequestObjectEncryptionMethod());
```

Comment:
lets remove the ternary conditions and populate the algorithm and method 1st and then pass here. That will improve readability

Reply:
Fixed in 1f4de9bb1b4a721922a631b048ee16397e81baea